### DISCUSSION TITLE
Which seeds need to be modified to generate completely new events.

### DISCUSSION THREAD

**Discussion by usert5432** [LS4GAN/toyzero#8] [Q&A]
Currently, toyzero package has two settings that control seeds:
```
Snakefile:112                 seed = "1,2,3,4"
```
and
```
cfg/main-depos-sim-adc.jsonnet:25 local seeds = [0,1,2,3,4];  // maybe let CLI set?
```

Could you please clarify which of those seeds need to be modified to generate a completely new event? In particular, is modifying the Snakefile seeds alone enough for completely new event generation?

**Comment by brettviren** [comment]
The first seed controls Python's random generator.  That stream is used to generate the directions and positions of the line tracks in the process of producing the "depos".  Changing this seed will result in the most obvious newness.

The second seed controls the random generator used in `wire-cell` (the C++ app).  It will control "everything else" in the simulation.  Changes in the random stream here may not be very noticeable but would include at least these things I can think of:

* the integral-preserving fluctuations applied across the binned Gaussian distribution of each depo.  
* the integral of that distribution itself depends on electron absorption (function of drift length) and is also fluctuated (based on the number of electrons absorbed).

**Comment by brettviren** [comment]
>  is modifying the Snakefile seeds alone enough for completely new event generation?

I would say "yes".  Simply generating new tracks will lead to different utilization of the C++ pRNG stream.  So, even with its seeds kept unchanged the fluctuations they provide will be unique.

**Comment by usert5432** [comment]
Perfect! Thank you!
